### DISCUSSION TITLE
How to switch model to "train=False" mode?

### DISCUSSION THREAD

**Discussion by PgLoLo** [google/flax#1561] [Q&A]
Consider a simple encoder-decoder model that can be parameterized by econder and decoder architecture:

```python
class EncoderDecoder(linen.Module):
    endoer: linen.Module
    decoder: linen.Module

    @linen.compact
    def __call__(self, x):
        return decoder(encoder(x))
```

Encoder/decoder layers may contain parts that can work differently during training and testing phases (like Dropout, Batchnorms, etc). But in examples on the internet, the only way I saw how to turn off Dropout is to manually pass the `train` argument to `EncoderDecoder` and then manually forward it into `Dropout` `deterministic argument.

But we can not do it for the case when we do not know the internal structure of the submodules (e.g. Dropout has an argument `deterministic` and Batchnorm has an argument `use_running_average`). Is there any approach closer to the PyTorch way of switching model training mode?
